templates/react/src/libs/providers.ts

```
@@ -109,12 +108,32 @@ async function sendTransactionViaExtension(
 async function sendTransactionViaWallet(
   transaction: ethers.providers.TransactionRequest
 ): Promise<TransactionState> {
-  transaction.value = BigNumber.from(transaction.value)
-  const res = await wallet.sendTransaction(transaction)
-  const receipt = await res.wait()
+  if (transaction.value) {
+    transaction.value = BigNumber.from(transaction.value)
+  }
+  const txRes = await wallet.sendTransaction(transaction)
+
+  let receipt = null
+  const provider = getProvider()
+  if (!provider) {
+    return TransactionState.Failed
+  }
```

Should we check this first to fail fast?